### ISSUE TITLE
custom_stoploss logic

### ISSUE THREAD

**Issue by Yaroslav-Koretskyi-BA** [freqtrade/freqtrade#7025]
## Describe your environment

  * Operating system: Mac OS
  * Python Version: Python 3.10.2 (`python -V`)
  * CCXT version: ccxt==1.89.14 (`pip freeze | grep ccxt`)
  * Freqtrade Version: freqtrade develop-3c1380fbc (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question
Hello, could you please explain a little bit more logic for custom_stoploss function:
for example, my custom_stoplos is: 

```
stoploss = -0.99
use_custom_stoploss=True
stoploss_price = 19599.775
trade.open_rate = 18711.00000000
```

```
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        if current_time - timedelta(minutes=5) > trade.open_date_utc:
            dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
            trade_date = timeframe_to_prev_date(self.timeframe, trade.open_date_utc)
            trade_candle = dataframe.loc[dataframe['date'] == trade_date]
            if not trade_candle.empty:
                stoploss_price = trade_candle['stoploss_price'].iloc[0]
                return stoploss_from_absolute(stoploss_price, trade.open_rate, is_short=trade.is_short)
```

so, I expect that stop loss will be changed every iteration, and if the current_rate will be equal to stoploss price it will return 0, and that would trigger an exit. Or it should be set once per trade, and if current_profit reaches stoploss, stoploss will be triggered?
```
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:

        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        open_candle = dataframe.loc[dataframe['date'] == trade.open_date_utc].squeeze()
        stoploss_price = open_candle['stoploss_price']
        if not open_candle.empty:
            if trade.is_short:
                return 1 - (stoploss_price/trade.open_rate)
            else:
                return (stoploss_price/trade.open_rate) - 1 

```
![Screenshot 2022-07-01 at 11 10 07](https://user-images.githubusercontent.com/71146654/176853590-9f4cb952-61b7-4e05-b87b-83feb8366028.png)

found in logs:
`2022-07-01 15:07:14,310 - freqtrade.persistence.trade_model - DEBUG - BTC/USDT - Stoploss adjusted. current_price=19177.10000000, open_rate=19462.50000000, max_rate=19487.30000000, initial_stop_loss=21408.75000000, stop_loss=19889.27200000. Trailing stoploss saved us: -1519.47800000.
`

so this is mean:
initial_stop_loss - trailing stoploss = stop_loss, right?

**Comment by xmatthias** [comment]
Custom stoploss works identical to trailing stoploss.

What you return is the relative offset to the current price - so assuming you return `-0.02` - stoploss would be trailing at 2%.

stoploss will however only be adjusted if current rate is raising (assuming a long) - so it'll trail - and then remain at the previously set value if current rate is falling again.
Obviously for longs this logic is reversed (down instead of up) - but the principle will remain the same

returning 0 will be seen as invalid response / no action - as you can't technically have a stoploss that's at current price (assuming you use stoploss on exchange, the exchange would reject that order with "would trigger immediately").
If you need to exit "now" - please use `custom_exit()` instead - as that's then no longer a stoploss scenario.
